src/main/java/com/kenny/openimgur/widget/GalleryWidgetRemoteViewsService.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2011 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.kenny.openimgur.widget;
+
+import android.appwidget.AppWidgetManager;
+import android.content.Intent;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.widget.RemoteViews;
+import android.widget.RemoteViewsService;
+
+import com.kenny.openimgur.R;
+import com.kenny.openimgur.activities.SettingsActivity;
+import com.kenny.openimgur.api.ApiClient;
+import com.kenny.openimgur.api.Endpoints;
+import com.kenny.openimgur.classes.ImgurAlbum;
+import com.kenny.openimgur.classes.ImgurBaseObject;
+import com.kenny.openimgur.classes.ImgurFilters;
+import com.kenny.openimgur.classes.ImgurPhoto;
+import com.kenny.openimgur.classes.OpenImgurApp;
+import com.kenny.openimgur.util.FileUtil;
+import com.kenny.openimgur.util.LogUtil;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GalleryWidgetRemoteViewsService extends RemoteViewsService {
+    @Override
+    public RemoteViewsFactory onGetViewFactory(Intent intent) {
+        return new GalleryWidgetRemoteViewsFactory((OpenImgurApp) getApplication(), intent);
```

Comment:
Use [OpenImageApp.getInstance](https://github.com/Kennyc1012/OpenImgur/blob/master/src/main/java/com/kenny/openimgur/classes/OpenImgurApp.java#L108) methods instead of type casting